### DISCUSSION TITLE
Running a testnet with telemetry on

### DISCUSSION THREAD

**Discussion by aesedepece** [witnet/witnet-rust#1962] [Ideas]
Ever since mainnet was launch, the idea of having a testnet for testing protocol changes and integratons without bloating or even risking the stability of the mainnet is something that has been suggested many times.

A testnet would be specially useful for early detection of corner cases and stuff if we enable telemetry in our nodes (e.g. Sentry), so that panics, errors and their context can be recorded and studied after-the-fact, without having to dissect unfriendly log traces.

This would be specially interesting for rehearsing network upgrades in the context of TAPI.

**Comment by drcpu-github** [comment]
That sounds like a pretty good idea, both for testing out changes in the protocol and for tool developers to try out stuff without needing mainnet Witnet coins.

**Reply by aesedepece** [reply]
I'm happy to put some testnet coins into a faucet :stuck_out_tongue:

**Reply by drcpu-github** [reply]
How generous of you!
